lib/accounting/config.rb

```diff
@@ -13,8 +13,18 @@ def initialize
     end
 
     def api_creds=(api_creds)
-      @api_creds = api_creds.deep_symbolize_keys
-      @api_creds.stringify_keys!
+      raise ArgumentError, 'Authnet API creds should be Proc or Hash' unless @api_creds.respond_to?(:call) || @api_creds.is_a?(Hash)
```

Comment:
This is one of those use cases where a statically typed language with a compiler would come in handy.